fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java

```diff
@@ -1158,6 +1181,41 @@ public TStreamLoadPutResult streamLoadPut(TStreamLoadPutRequest request) {
         return result;
     }
 
+    private void streamLoadPutWithSqlImpl(TStreamLoadPutRequest request) throws UserException {
+        String loadSql = request.getLoadSql();
+        ConnectContext ctx = new ConnectContext(null);
+        ctx.setEnv(Env.getCurrentEnv());
+        ctx.setQueryId(request.getLoadId());
+        ctx.setCluster(SystemInfoService.DEFAULT_CLUSTER);
+        ctx.setCurrentUserIdentity(UserIdentity.ROOT);
+        ctx.setQualifiedUser(UserIdentity.ROOT.getQualifiedUser());
+        ctx.setThreadLocalInfo();
+        ctx.setBackendId(request.getBackendId());
+        ctx.setLoadId(request.getLoadId());
+        SqlScanner input = new SqlScanner(new StringReader(loadSql), ctx.getSessionVariable().getSqlMode());
+        SqlParser parser = new SqlParser(input);
+        try {
+            StatementBase parsedStmt = SqlParserUtils.getFirstStmt(parser);
```

Comment:
should not call coord.exec() in rpc thread context. It will use up all rpc threads.

Reply:
should add the coord in to a map<load id, coord> and use a thread pool to check coord status and send status to related be.